pkg/network/protocols/events/consumer_test.go

```diff
@@ -78,6 +78,33 @@ func TestConsumer(t *testing.T) {
 	}
 }
 
+func TestInvalidBatchCountMetric(t *testing.T) {
+	kversion, err := kernel.HostVersion()
+	require.NoError(t, err)
+	if minVersion := kernel.VersionCode(4, 14, 0); kversion < minVersion {
+		t.Skipf("package not supported by kernels < %s", minVersion)
+	}
+
+	program, err := newEBPFProgram(config.New())
+	require.NoError(t, err)
+
+	consumer, err := NewConsumer("test", program, func([]uint64) {})
+	require.NoError(t, err)
+
+	consumer.handler.(*ddebpf.RingBufferHandler).RecordHandler(&ringbuf.Record{
+		RawSample: []byte("test"),
+	}, nil, nil)
+
+	consumer.invalidBatchCount.Reset()
+	consumer.Start()
+	require.Eventually(t, func() bool {
+		program.Stop(manager.CleanAll)
+		return true
+	}, 1*time.Second, 100*time.Millisecond, "failed to stop program")
+
+	require.Equalf(t, 1, int(consumer.invalidBatchCount.Get()), "invalidBatchCount should be equal to 1")
```

Comment:
what in the test makes the batch invalid?

Reply:
The length of the data is 4, and it is smaller than the size of the batch.

Reply:
how do you expect anyone to understand that from the current code?

Reply:
Added documentation

Reply:
where?

Reply:
Pushed in the latest commit :)